### PR TITLE
66 comma and spaces

### PR DESCRIPTION
**Pull request by Yannis-Hofmann**
- Adding config with option to switch between styles:
- Space before comma and no space before comma
- Space before dot in array and no space before dot in array

This config will be used for all future configuration settings/prefferences.

- Adding test which covers this new option

- Timeout of testAllMethods increased

### CODE REVIEW

File: packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st [hunk 1]
```diff
@@ -0,0 +1,4 @@
+helper
+getConfigSpacesBeforePointinArray
```
Comment (tom95) [inline packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st:2]: `Pointin` vs `PointIn`

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/config..st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+accessing
+config: aPPFormatterConfig
+
+	config := aPPFormatterConfig.
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/config..st:4]: trailing dot

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/initialize.st [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 initialize
-
 	super initialize.
 	stream := '' writeStream.
 	indent := 0.
 	preFormatCache := Dictionary new.
-	comments := OrderedCollection new
\ No newline at end of file
+	comments := OrderedCollection new.
+	config := PPFormatterConfig default.
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/initialize.st:8]: trailing dot

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/printBinaryMessage.multiLine..st [hunk 4]
```diff
@@ -1,8 +1,8 @@
 printing
 printBinaryMessage: aNode multiLine: multiLine
 
-	aNode originalSelector ~= #, ifTrue: [stream space].
-	stream
+	(aNode originalSelector ~= #, or: [config spaceBeforeComma]) ifTrue: [stream space].
+	stream
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/printBinaryMessage.multiLine..st:5]: trailing whitespace

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/stream..st [hunk 5]
```diff
@@ -0,0 +1,4 @@
+accessing
+stream: anObject
+
+	stream := anObject.
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/stream..st:4]: trailing dot

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st [hunk 6]
```diff
@@ -10,12 +10,13 @@ visitBraceNode: aNode
...
 		
-		aNode elements do: [:element | | isLast |
-			isLast := element = aNode elements last.
-			
+		aNode elements do: [:element | | isLast |				
+			isLast := element = aNode elements last.			
 			(self isEmptyLineMarker: element) ifFalse: [
-				self visitNode: element.
-				isLast ifFalse: [stream nextPut: $.]].
+
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st:16]: looks like an erroneous newline?

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st [hunk 7]
```diff
@@ -10,12 +10,13 @@ visitBraceNode: aNode
...
-			isLast := element = aNode elements last.
-			
+		aNode elements do: [:element | | isLast |				
+			isLast := element = aNode elements last.			
 			(self isEmptyLineMarker: element) ifFalse: [
-				self visitNode: element.
-				isLast ifFalse: [stream nextPut: $.]].
+				
+				self visitNode: element.				
+				isLast ifFalse: [config spaceBeforePointInArray ifTrue: [stream space].  stream nextPut: $.]].
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st:18]: looks a bit crammed, I think you could give those two statements two lines as well :D

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitReturnNode..st [hunk 8]
```diff
@@ -2,4 +2,5 @@ visiting
 visitReturnNode: aNode
 
 	stream nextPutAll: '^ '.
-	self visitNode: aNode expr
\ No newline at end of file
+	self visitNode: aNode expr.
+
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitReturnNode..st:6]: trailing newline and dot

File: packages/PoppyPrint-Core.package/PPFormatterConfig.class/class/default.st [hunk 9]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
+default
+ "Default is lazy initialized"
+ ^ self new.
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatterConfig.class/class/default.st:4]: trailing dot

File: packages/PoppyPrint-Core.package/PPFormatterConfig.class/instance/spaceBeforeComma.st [hunk 10]
```diff
@@ -0,0 +1,5 @@
+accessing
+spaceBeforeComma
+
+	^ spaceBeforeComma ifNil: [spaceBeforeComma := false].
+
```
Comment (tom95) [inline packages/PoppyPrint-Core.package/PPFormatterConfig.class/instance/spaceBeforeComma.st:5]: trailing newline and dot

**Review by tom95 — CHANGES_REQUESTED** [review]
Looks great! Two notes:

* I added a couple comments for nitpicking style/whitespace
* one edge case you may want to add a test for is when an array is displayed on multiple lines

Concerning the style: I would suggest that you fix the noted places manually and then create another PR for master where you apply poppyprint to the entire codebase. Afterwards, all future PRs can just be poppy-printed as well. Poppy-printing this PR will likely result in a number of changes that are unrelated to the actual PR, so that's why I would do that separately.

File: packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st (hunk 1 above)
Comment (kdauer) [reply packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st:2]: done

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/initialize.st (hunk 3 above)
Comment (kdauer) [reply packages/PoppyPrint-Core.package/PPFormatter.class/instance/initialize.st:8]: done

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/printBinaryMessage.multiLine..st (hunk 4 above)
Comment (kdauer) [reply packages/PoppyPrint-Core.package/PPFormatter.class/instance/printBinaryMessage.multiLine..st:5]: done

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/stream..st (hunk 5 above)
Comment (kdauer) [reply packages/PoppyPrint-Core.package/PPFormatter.class/instance/stream..st:4]: done

File: packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st (hunk 7 above)
Comment (kdauer) [reply packages/PoppyPrint-Core.package/PPFormatter.class/instance/visitBraceNode..st:18]: done

File: packages/PoppyPrint-Core.package/PPFormatterConfig.class/class/default.st (hunk 9 above)
Comment (kdauer) [reply packages/PoppyPrint-Core.package/PPFormatterConfig.class/class/default.st:4]: done

**Comment by kdauer** [comment]
The edge case test for multiline arrays will be implemented in #50

File: packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st (hunk 1 above)
Comment (kdauer) [reply packages/PoppyPrint-Tests.package/PPFormatterTest.class/instance/getConfigSpacesBeforePointinArray.st:2]: > * one edge case you may want to add a test for is when an array is displayed on multiple lines

The edge case test for multiline arrays will be implemented in https://github.com/hpi-swa-teaching/poppy-print/issues/50
